### ISSUE TITLE
Client server split imposes unnecessary restrictions

### ISSUE THREAD

**Issue by christazelaar** [voxpupuli/puppet-rsyslog#122]
We recently switched over to this module from saz/rsyslog (while it was marked as deprecated) and ran into an issue.

To avoid confusion - i will call physical servers "machines" so we can use "client" and "server" only to refer to the module usage.

Our use case is that we have multiple data centers, and in each one we have a couple of machines that (among other things) collect all the logs from all the other machines in the data center (including each other). So in any data center, all our machines are clients. But two machines are _also_ servers. 

Since every machine in our infrastructure is a client, I started out using rsyslog::client at a level of the hierarchy which covers all machines (as we did with saz/rsyslog). My syslog collection server is in its own module, so I can tag those servers as needing that module and get them writing logs to file. 

I mistakenly thought I would be able to use rsyslog::server to configure the two machines per datacenter. Nope. ` Duplicate declaration: Class[Rsyslog::Config] is already declared in file ...`

So then I tried to hack it by passing my rulesets etc. to the client class ... which you cant #58 
So then I tried to look for an equivalent of saz/rsyslog::snippet ... also no joy. #109 

So that leaves me with the options of:
- Use rsyslog::server for all clients (confusing)
- Write a server module that drops some files into /etc/rsyslod.d/ (poor cohesiveness) 

The crux of it:
The design choice of forcing separation of clients and servers (which for a lot of people is convenient) ends up imposing a restriction on the user that the underlying rsyslog software does not. 

The best solution would be one where I can configure rsyslog using hiera, or in a module,  without restriction. The server/client split could be dropped completely in that scenario, as I could define that myself.

In lieu of that, implementing #109 would help tremendously in the short term. This functionality is tremendously powerful and useful. 

Thanks for your time! I hope this feedback is useful in further improving this module. 

👍 on the documentation BTW :)

**Comment by dhollinger** [comment]
I can look into removing the client/server split. It was already there when we inherited the module.

The way I had been using it before we took over maintenance was that if a _machine_ was functioning as an rsyslog server, then all that machine's rsyslog config is configured as a server. The client side was for _machines_ that are purely operating as clients and not ever as servers.

That said, there is (as far as I know) no reason to maintain the client/server split.
